pkg/storage/utils_test.go

```diff
@@ -0,0 +1,192 @@
+package storage
+
+import (
+	"reflect"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestGetMachineReadableFormat(t *testing.T) {
+	tests := []struct {
+		name         string
+		storageName  string
+		storageSize  string
+		mountedPath  string
+		activeStatus bool
+		want         Storage
+	}{
+		{
+			name:         "test case 1: with a pvc, valid path and mounted status",
+			storageName:  "pvc-example",
+			storageSize:  "100Mi",
+			mountedPath:  "data",
+			activeStatus: true,
+			want: Storage{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pvc-example",
+				},
+				TypeMeta: metav1.TypeMeta{Kind: "storage", APIVersion: "odo.dev/v1alpha1"},
+				Spec: StorageSpec{
+					Size: "100Mi",
+					Path: "data",
+				},
+			},
+		},
+		{
+			name:         "test case 2: with a pvc, empty path and unmounted status",
+			storageName:  "pvc-example",
+			storageSize:  "500Mi",
+			mountedPath:  "",
+			activeStatus: false,
+			want: Storage{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pvc-example",
+				},
+				TypeMeta: metav1.TypeMeta{Kind: "storage", APIVersion: "odo.dev/v1alpha1"},
+				Spec: StorageSpec{
+					Size: "500Mi",
+					Path: "",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotStorage := GetMachineReadableFormat(tt.storageName, tt.storageSize, tt.mountedPath)
+			if !reflect.DeepEqual(tt.want, gotStorage) {
+				t.Errorf("the returned storage is different, expected: %v, got: %v", tt.want, gotStorage)
+			}
+		})
+	}
+}
+
+func TestGetMachineReadableFormatForList(t *testing.T) {
+
+	tests := []struct {
+		name         string
+		inputStorage []Storage
+		want         StorageList
+	}{
+		{
+			name: "test case 1: with a single pvc",
+			inputStorage: []Storage{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "example-pvc-1",
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "List",
```

Comment:
Shouldn't the kind be "storage"?  Same for below cases.

Reply:
"List" indicates that the output would be of a list of certain resources. It's the same when list of kubernetes resources are returned.

Reply:
I understand that the value of higher level '.kind' is 'List' when returning a list of resources, but the lower lever '.items[*].kind' must have a value corresponding to the kind of resource it is, in this case 'Storage'. Isn't it?

Reply:
> but the lower lever .items[*].kind must have a value corresponding to the kind of resource it is, in this case 'Storage'. Isn't it?

Yes

Reply:
> > but the lower lever .items[*].kind must have a value corresponding to the kind of resource it is, in this case 'Storage'. Isn't it?
> 
> Yes

So, should it be 'Storage' instead of 'List'?

Reply:
At the top level it's `List` and the inner resources are `Storage`.